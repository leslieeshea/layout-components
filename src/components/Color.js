import React from 'react';

export default function Color() {
  const color = {
    name: 'Pearl Violet',
    hex: '#8673A1',
    rgb: {
      red: 134,
      green: 115,
      blue: 161
    }
  };

  return (
    <dl>
      <dt>Name</dt>
      <dd>{color.name}</dd>

      <dt>Hex</dt>
      <dd>{color.hex}</dd>

      <dt>RGB</dt>
      <dd>Red - {color.rgb.red}</dd>
      <dd>Green - {color.rgb.green}</dd>
      <dd>Blue - {color.rgb.blue}</dd>
    </dl>
  );
}
