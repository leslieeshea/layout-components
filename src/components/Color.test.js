import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color Component', () => {
  it('renders a Color', () => {
    const wrapper = shallow(<Color />);
    expect(wrapper).toMatchSnapshot();
  });
});
