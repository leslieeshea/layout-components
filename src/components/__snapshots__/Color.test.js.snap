// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Component renders a Color 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd>
    Pearl Violet
  </dd>
  <dt>
    Hex
  </dt>
  <dd>
    #8673A1
  </dd>
  <dt>
    RGB
  </dt>
  <dd>
    Red - 
    134
  </dd>
  <dd>
    Green - 
    115
  </dd>
  <dd>
    Blue - 
    161
  </dd>
</dl>
`;
